### PR TITLE
Generalize ActionSheetManager items URL 

### PR DESCRIPTION
**Pull request by fxfactorial**
Not ready but want to start conversation.  (i.e. not sure if to change key name to `urls` from `url`) 

I want to be able to share more than 1 image at a time, the current code only lets one use one url, but the API allows for much more. 

Some places already make this an issue: 

https://github.com/EstebanFuentealba/react-native-share/issues/85

And I also need to be able to email/send on iMessage, etc with multiple URL resources.

### CODE REVIEW

**Comment by fxfactorial** [comment]
And I've just tested this locally in my project and its works correctly.

**Comment by fxfactorial** [comment]
@hramos Ping, can someone take a look at this please?

**Comment by hramos** [comment]
Sorry @fxfactorial, we do have quite the backlog of PRs. Can you think of anyone that has worked on this file before that may be able to review the changes?

**Comment by fxfactorial** [comment]
@hramos Sure, I understand that RN is a big project, and probably people are overworked, thank you for looking at all. 

I think this would be like a couple of moments of @nicklockwood 's time to glance over.

**Comment by fxfactorial** [comment]
@hramos another friendly ping

**Comment by hramos** [comment]
@fxfactorial hey, it looks like this has introduced an issue. Here's a simple repro:

```
require('ActionSheetIOS').showShareActionSheetWithOptions(
  {message: 'TEST!'},
  // or {url: 'http://google.com'},
  (error) => {},
  () => {},
);
```

Results in a red screen.

File: Libraries/ActionSheetIOS/RCTActionSheetManager.m [hunk 1]
```diff
@@ -133,11 +133,17 @@ - (CGRect)sourceRectInView:(UIView *)sourceView
   if (message) {
     [items addObject:message];
   }
-  NSURL *URL = [RCTConvert NSURL:options[@"url"]];
-  if (URL) {
-    if ([URL.scheme.lowercaseString isEqualToString:@"data"]) {
+  NSArray<NSURL *> *URLS = [RCTConvert NSURLArray:options[@"url"]];
+
+  if (URLS.count == 0) {
```
Comment (olegbl) [inline Libraries/ActionSheetIOS/RCTActionSheetManager.m]: This means that passing in "message" will always crash.

**Comment by hramos** [comment]
We're going to back out this breaking change.

File: Libraries/ActionSheetIOS/RCTActionSheetManager.m [hunk 2]
```diff
@@ -133,11 +133,17 @@ - (CGRect)sourceRectInView:(UIView *)sourceView
   if (message) {
     [items addObject:message];
   }
-  NSURL *URL = [RCTConvert NSURL:options[@"url"]];
-  if (URL) {
-    if ([URL.scheme.lowercaseString isEqualToString:@"data"]) {
+  NSArray<NSURL *> *URLS = [RCTConvert NSURLArray:options[@"url"]];
```
Comment (olegbl) [inline Libraries/ActionSheetIOS/RCTActionSheetManager.m]: This is a breaking API change that introduces crashes into all existing apps using it. Highly suggest introducing a "urls" argument instead.

**Pull request closed without merging**